### PR TITLE
[issue-2187] Type declarations for src/Collection

### PR DESCRIPTION
**Pull request by bsacharski**
- [X] Improvement for #2187 (part of work)
- [ ] New feature
- [ ] BC breaks
- [ ] Deprecations
- [ ] Tests added
- [X] Changelog updated

Refactored the code in `src/Collection` to include type declarations for method arguments and returned values.
Also updated the code in `src/Configuration/Configuration.php` to add missing type declaration for `ownValues`

### CODE REVIEW

File: src/Collection/Collection.php [hunk 1]
```diff
@@ -14,11 +14,15 @@ class Collection implements Countable, IteratorAggregate
...
 
-    public function all()
+    public function all(): array
     {
         return $this->values;
     }
 
+    /**
+     * @return mixed|void
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingNativeTypeHint
```
Comment (antonmedv) [inline src/Collection/Collection.php:24]: Is there a way to opt-out from this check? This looks like very verbose stuff.

Comment (bsacharski) [reply src/Collection/Collection.php:24]: Given [docs](https://github.com/slevomat/coding-standard#slevomatcodingstandardtypehintsreturntypehint-) none that I am aware of.

**Comment by antonmedv** [comment]
Please, rebase.

File: src/Collection/Collection.php (hunk 1 above)
Comment (antonmedv) [reply src/Collection/Collection.php:24]: I found a way. Please, drop those lines.

Comment (bsacharski) [reply src/Collection/Collection.php:24]: Will do. Can you share details how you're doing that?

Comment (bsacharski) [reply src/Collection/Collection.php]: Ok, I see how you've added configuration setting to phpcs.xml. Didn't know about that one :+1:

File: src/Task/TaskCollection.php [hunk 2]
```diff
@@ -15,7 +15,7 @@
  */
 class TaskCollection extends Collection
 {
-    protected function throwNotFound(string $name)
+    protected function throwNotFound(string $name): void
```
Comment (antonmedv) [inline src/Task/TaskCollection.php:18]: So, is this `: void` parts also necessary? 
We have lots and lots of anon function in recipes, do we really want to add it there? What benefits will it provide? 
Should we omit recipes/ all together?

Comment (bsacharski) [reply src/Task/TaskCollection.php:18]: I would skip recipes, however IMO the core library should contain as much of native and phpDoc-based type definitions as possible.

As for void being hinted, IMO yes. This having explicit void means that function should never, ever, return anything. This in turn helps with finding bugs even on static analysis level, as PHPStan will report an issue if a piece of code tries to use a void result.

**Pull request merged**